src/simix/smx_synchro.cpp

```diff
@@ -356,19 +358,34 @@ void SIMIX_cond_broadcast(smx_cond_t cond)
  * Destroys and frees the condition's memory. 
  * \param cond A condition
  */
-void SIMIX_cond_destroy(smx_cond_t cond)
+void SIMIX_cond_unref(smx_cond_t cond)
 {
   XBT_IN("(%p)",cond);
   XBT_DEBUG("Destroy condition %p", cond);
-
   if (cond != nullptr) {
+    intrusive_ptr_release(cond);
+  }
+  XBT_OUT();
+}
+
+
+void intrusive_ptr_add_ref(s_smx_cond_t *cond)
```

Comment:
I would to but the header in which s_smx_cond_t is defined is private which lead to undefined references. Maybe could I put it into simix.h ?